crates/bin/node/src/grpc_service.rs

```diff
@@ -57,9 +71,23 @@ impl GrpcState {
             signer: signer_client,
             #[cfg(feature = "starknet")]
             starknet_cashier,
+            response_cache: Arc::new(InMemResponseCache::new(None)),
         }
     }
 
+    pub async fn should_cache_endpoint<M: nuts::traits::Method + Display + FromStr>(
```

Comment:
are you not calling this only when you already know you should be caching?

Is there cases where it is not clear at compile time that you should be caching or not?